rudder-core/src/main/scala/com/normation/rudder/batch/UpdateDynamicGroups.scala

```diff
@@ -174,76 +169,72 @@ class UpdateDynamicGroups(
       // This case is launched when an update was pending, it only launch the process
       // and it does not schedule a new update.
       case DelayedUpdate =>
+        onePending = false
         processUpdate
+
       //
       //Process a dynamic group update response
       //
-      case DynamicUpdateResult(id, modId, start,end,results) => //TODO: other log ?
-        logger.trace("***** Get result for process: " + id)
+      case DynamicUpdateResult(id, modId, start, end, results) => //TODO: other log ?
+        logger.trace(s"***** Get result for process: ${id}")
 
-        currentState = IdleUdater
+        currentState = IdleGroupUpdater
 
-        var retriggerDeployment = false
+        // If one update is pending, immediately start a new group update
+        // We should no deploy during this run, but we still need to compute if we need a deployment for this group update
+        // Deployment will be launched when the pending group update will be finished (and even if it does not need any deployment)
+        val delayDeployment = onePending
 
-        // If one update is pending, immediatly start one other
-        // It should not clear the needDeployement
+        // Maybe should be done at the end, when we know it's ok to start deployment ...
         if(onePending) {
-          onePending = false
```

Comment:
i'm pretty sure this will break https://www.rudder-project.org/redmine/issues/10446

Reply:
Kept the logic of retrigger deployment, which prevents policy generation if there is a pending update 

Removed local var never reassigned in scope replaced by a val

Removed a needdeployment set to false, which only make you lose a deployment if last update does not change anything 

Even if needdeployment is true, it will be prevented by retrigger deployment or the updater not being idle

Can you point me something that would break it ? I will happily look into it 

I'll also try on Monday

Reply:
Could you share the test results? The PR seems ok.

Reply:
I tested to accepted a hundred of Nodes with 4-5  dynamic groups with regexp.
Right All Nodes are accepted and generation starts, it appears that all groups contains all nodes they should contain (if running search, no more nodes appear, same when running group update)

And only one generation is started (after all Nodes accepted)